### PR TITLE
#1534 - fix warning proposal

### PR DESCRIPTION
**Pull request by btry**
This is a proposal for discussion

### CODE REVIEW

File: inc/db.function.php [hunk 1]
```diff
@@ -1705,15 +1705,17 @@ function getEntitiesRestrictRequest($separator="AND", $table="", $field="",$valu
...
 
-   $query .= "$field";
-
    if (is_array($value)) {
-      $query .= " IN ('" . implode("','",$value) . "') ";
+      $query .= "$field IN ('" . implode("','",$value) . "') ";
    } else {
       if (strlen($value) == 0) {
-         $query .= " IN (".$_SESSION['glpiactiveentities_string'].") ";
+         if (isset($_SESSION['glpiactiveentities_string'])) {
```
Comment (trasher) [inline inc/db.function.php]: I do not thing this is the way to work...

I guess that is entities string is not set; user is not logged correctly, and his entities are not loaded; I'd prefer the auth side to be fixed.

If entities are not loaded, there will be no restrictions at all, and user may show things from entities he's not part of.

**Comment by remicollet** [comment]
IMHO, if you raise this warning, a call relying on the session without the session, you have an issue in the calling stack which need to be fixed.

Perhaps you should even move this warning into a "critical" error ;)

Trying to make this warning silent doesn't sounds like a good idea to me.

**Comment by btry** [comment]
For reference : #1534 

In a discussion with @orthagh we noticed 2 things  

- It seems before this commit 2fab429 user changes happening in Auth::login() where not logged (see history tab of the user).

- without this fix, a warning occurs, and the generated SQL is invalid, but there is no error in sql-errors.log, because the SQL is simply not used while authenticating the user : it is stored in a search option dynamically generated in link.class.php. 

I deduce the updates on a user are logged by a side effect of 2fab429 . Then we may consider disable logging in Auth::login() .
https://github.com/glpi-project/glpi/blob/master/inc/auth.class.php#L774

However, I found interesting to keep track of events of a user being disabled due to vanish of his LDAP entry.

Also, I wonder if there us a use case where getEntitiesRestrictRequest() is called and no user is logged in.

**Comment by btry** [comment]
Replaced by #1627

**Pull request closed without merging**